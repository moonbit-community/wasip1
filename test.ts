import { assertEquals } from '@std/assert';
import { assertSnapshot } from '@std/testing/snapshot';

const textDecoder = new TextDecoder('utf-8');

Deno.test("arg_get", async (t) => {
    const output = await new Deno.Command("wasmtime", {
        args: ['--argv0', 'args_get.wasm', 'test/target/wasm/debug/build/args_get/args_get.wasm'],
        stdout: 'piped',
        stderr: 'piped',
    }).spawn().output()
    assertEquals(output.code, 0);
    const stdout = textDecoder.decode(output.stdout)
    await assertSnapshot(t, stdout);
})

Deno.test("environ_get", async (t) => {
    const output = await new Deno.Command("wasmtime", {
        args: ["--env", "hello=world", 'test/target/wasm/debug/build/environ_get/environ_get.wasm'],
        stdout: 'piped',
        stderr: 'piped',
    }).spawn().output()
    assertEquals(output.code, 0);
    const stdout = textDecoder.decode(output.stdout)
    await assertSnapshot(t, stdout);
})

Deno.test("fd_write", async (t) => {
    // Notice that wasmtime may not write everything required in the buffer.
    const output = await new Deno.Command("wasmtime", {
        args: ['test/target/wasm/debug/build/fd_write/fd_write.wasm'],
        stdout: 'piped',
        stderr: 'piped',
    }).spawn().output()
    assertEquals(output.code, 0);
    const stdout = textDecoder.decode(output.stdout)
    await assertSnapshot(t, stdout);
})