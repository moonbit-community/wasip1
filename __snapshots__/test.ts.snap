export const snapshot = {};

snapshot[`arg_get 1`] = `""`;

snapshot[`environ_get 1`] = `""`;

snapshot[`fd_fdstat_get 1`] = `
"{fs_filetype: Unknown, fs_flags: FdFlag(0), fs_rights_base: Right(64), fs_rights_inheriting: Right(64)}
{fs_filetype: Directory, fs_flags: FdFlag(0), fs_rights_base: Right(129498624), fs_rights_inheriting: Right(267911167)}
"
`;

snapshot[`fd_write 1`] = `
"Hello
Size(6)"
`;
